source/IntegrationTests/CimMessageAdapter/Messages/RequestAggregatedMeasureData/RequestAggregatedMeasureDataReceiverTests.cs

```diff
@@ -0,0 +1,558 @@
+
+        Assert.Contains(result.Errors, error => error is AuthenticatedUserDoesNotMatchSenderId);
+    }
+
+    [Fact]
+    public async Task Authenticated_user_must_hold_the_role_type_as_specified_in_message()
+    {
+        await using var message = BusinessMessageBuilder
+            .RequestAggregatedMeasureData()
+            .Message();
+
+        var messageParserResult = await ParseMessageAsync(message).ConfigureAwait(false);
+        var result = await CreateMessageReceiver().ReceiveAsync(messageParserResult, CancellationToken.None).ConfigureAwait(false);
+
+        Assert.Contains(result.Errors, error => error is AuthenticatedUserDoesNotHoldRequiredRoleType);
+    }
+
+    [Fact]
+    public async Task Series_must_have_unique_transaction_ids()
+    {
+        var knownReceiverId = "5790001330552";
+        var knownReceiverRole = "DDZ";
+        await CreateIdentityWithRoles(new List<MarketRole> { MarketRole.EnergySupplier })
+            .ConfigureAwait(false);
+        await using var message = BusinessMessageBuilder
+            .RequestAggregatedMeasureData()
+            .WithReceiverRole(knownReceiverRole)
+            .WithReceiverId(knownReceiverId)
+            .WithSenderRole(MarketRole.EnergySupplier.Code)
+            .WithSenderId(SampleData.SenderId)
+            .DuplicateSeriesRecords()
+            .Message();
+
+        var messageParserResult = await ParseMessageAsync(message).ConfigureAwait(false);
+        var result = await CreateMessageReceiver().ReceiveAsync(messageParserResult, CancellationToken.None).ConfigureAwait(false);
+
+        Assert.Contains(result.Errors, error => error is DuplicateTransactionIdDetected);
+    }
+
+    [Fact]
+    public async Task Series_must_have_none_unique_transaction_ids_across_senders()
+    {
+        var knownReceiverId = "5790001330552";
+        var knownReceiverRole = "DDZ";
+        await CreateIdentityWithRoles(new List<MarketRole> { MarketRole.EnergySupplier })
+            .ConfigureAwait(false);
+        await using var message = BusinessMessageBuilder
+            .RequestAggregatedMeasureData()
+            .WithReceiverRole(knownReceiverRole)
+            .WithReceiverId(knownReceiverId)
+            .WithSenderRole(MarketRole.EnergySupplier.Code)
+            .WithSenderId(SampleData.SenderId)
+            .Message();
+
+        var messageParserResult = await ParseMessageAsync(message).ConfigureAwait(false);
+        var resultFromFirstMessage = await CreateMessageReceiver().ReceiveAsync(messageParserResult, CancellationToken.None).ConfigureAwait(false);
+
+        // Request from a second sender.
+        await using var message02 = BusinessMessageBuilder
+            .RequestAggregatedMeasureData()
+            .WithReceiverRole(knownReceiverRole)
+            .WithReceiverId(knownReceiverId)
+            .WithMessageId("123564755")
```

Comment:
```suggestion
            .WithMessageId("123564789123564789123564789123564789")
```
MessageId must have 36 chars